cscs-checks/mch/openacc_cuda_mpi_cppstd.py

```diff
@@ -46,9 +45,20 @@ def __init__(self):
 
     @run_before('compile')
     def setflags(self):
+        if (not self.current_environ.name.startswith('PrgEnv-nvidia')):
+            self.modules = ['craype-accel-nvidia60']
         if self.current_environ.name.startswith('PrgEnv-cray'):
             self.build_system.fflags += ['-hacc', '-hnoomp']
 
+        elif self.current_environ.name.startswith('PrgEnv-nvidia'):
+            self.build_system.fflags += ['-acc']
+            self.build_system.fflags += ['-ta:tesla:cc60']
+            self.build_system.ldflags = [
+                '-acc', '-ta:tesla:cc60', '-Mnorpath', '-lstdc++',
+                '-L/opt/nvidia/hpc_sdk/Linux_x86_64/21.3/cuda/11.2/lib64',
+                '-lcudart'
```

Comment:
I think you should just do `-Mcuda` in here.

Reply:
you are right, unfortunately the segfault remains

Reply:
looks more elegant, although I thought -Mcuda enables mainly cuda-Fortran

Reply:
Looks like they've changed it to just `-cuda` these days.